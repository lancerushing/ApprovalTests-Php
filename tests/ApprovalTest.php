<?php

class ApprovalTest extends PHPUnit_Framework_TestCase {
	private function approveList($list){
		Approvals::useReporter(new OpenDiffReporter());
		Approvals::approveList($list);
	}

	public function testApproveArray() {
		$list = array('zero', 'one', 'two', 'three', 'four', 'five');
		$this -> approveList($list);
	}

	public function testApproveMap() {
		$list = array(
			'zero' => 'Lance',
			'one' => 'Jim',
			'two' => 'James',
			'three' => 'LLewellyn',
			'four' => 'Asaph',
			'five' => 'Dana'
		);
		$this -> approveList($list);
	}

	public function testAppoveMultiArray(){
		$list = array(array("Hello", "World"), array("Tom", "Joe"));
		$this -> approveList($list);
	}

	public function testApproveMultiMap(){
		$list = array(
			'zero' => 'Lance',
			'one' => 'Jim',
			'two' => 'James',
			'three' => array('primary' => 'LLewellyn', 'other' => 'Jason')
		);
		$this -> approveList($list);
	}
}
?>
