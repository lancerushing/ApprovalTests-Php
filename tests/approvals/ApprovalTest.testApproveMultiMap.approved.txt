[zero] -> Lance
[one] -> Jim
[two] -> James
[three] -> 
	[primary] -> LLewellyn
	[other] -> Jason
